src/pygeom.c

```diff
@@ -230,20 +321,22 @@ static PyObject *GeometryObject_new(PyTypeObject *type, PyObject *args,
     }
     else if (PyUnicode_Check(value)) {
         return GeometryObject_FromWKT(value);
-    }
-    else {
+    } else if (PyBytes_Check(value)) {
+        return GeometryObject_FromWKB(value);
+    } else {
         PyErr_Format(PyExc_TypeError, "Expected string, got %s", value->ob_type->tp_name);
         return NULL;
     }
 }
 
 static PyMethodDef GeometryObject_methods[] = {
+    {"__reduce__", GeometryObject_reduce, METH_NOARGS, "For pickling."},
```

Comment:
I haven't had a chance to figure out why, but this is raising a compiler warning about incompatible pointer types:

```
src/pygeom.c:333:20: warning: incompatible pointer types initializing 'PyCFunction' (aka 'struct _object *(*)(struct _object *, struct _object *)') with an expression of type 'PyObject *(PyObject *)' (aka 'struct _object *(struct _object *)')
      [-Wincompatible-pointer-types]
    {"__reduce__", GeometryObject_reduce, METH_NOARGS, "For pickling."},
```